src/java/org/apache/cassandra/db/compaction/unified/Controller.java

```diff
@@ -1240,7 +1247,7 @@ private static long validateSizeWithAlt(Map<String, String> options, String opti
 
     private static void validateNoneWith(Map<String, String> options, String option, String... incompatibleOptions)
     {
-        if (!options.containsKey(option) || Arrays.stream(incompatibleOptions).noneMatch(options::containsKey))
+        if (Arrays.stream(incompatibleOptions).noneMatch(options::containsKey))
```

Comment:
I inlined the method as `option` is not in `options` when the check is performed.

Reply:
Yes, but I would keep the check even though it is redundant to make sure the method does what it says it does.

Alternatively the name should be changed, and the `option` parameter becomes harder to justify... it's easier to just add the check back.

Reply:
The method is now removed and and its content is a part of `validateOptions`. I wasn't able keep `!options.containsKey(option)` part of the predicate because it short-circuited the check (it was always `true`).